crates/proof-of-sql/src/proof_primitive/dory/public_parameters.rs

```diff
@@ -21,14 +23,33 @@ pub struct PublicParameters {
 }
 
 impl PublicParameters {
-    #[cfg(any(test, feature = "test"))]
+    /// Create new public parameters from the requisite group element vectors
+    /// and setup size.
+    pub fn new(
+        Gamma_1: Vec<G1Affine>,
+        Gamma_2: Vec<G2Affine>,
+        H_1: G1Affine,
+        H_2: G2Affine,
+        Gamma_2_fin: G2Affine,
+        max_nu: usize,
+    ) -> Self {
+        Self {
+            Gamma_1,
+            Gamma_2,
+            H_1,
+            H_2,
+            Gamma_2_fin,
+            max_nu,
+        }
+    }
+
```

Comment:
instead of changing the visibility of the fields from pub(super) we can just add a constructor here